From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 12 Oct 2023 22:08:52 -0500
Subject: [PATCH] aeryn login.defs changes

Change home mode for samba share, use yescrypt by default
---
 etc/login.defs | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/etc/login.defs b/etc/login.defs
index 33622c29..ce731062 100644
--- a/etc/login.defs
+++ b/etc/login.defs
@@ -158,8 +158,8 @@ ENV_HZ		HZ=100
 # *REQUIRED*  The default PATH settings, for superuser and normal users.
 #
 # (they are minimal, add the rest in the shell startup files)
-ENV_SUPATH	PATH=/sbin:/bin:/usr/sbin:/usr/bin
-ENV_PATH	PATH=/bin:/usr/bin
+# ENV_SUPATH	PATH=/sbin:/bin:/usr/sbin:/usr/bin
+# ENV_PATH	PATH=/bin:/usr/bin
 
 #
 # Terminal permissions
@@ -204,7 +204,9 @@ UMASK		022
 # HOME_MODE is used by useradd(8) and newusers(8) to set the mode for new
 # home directories.
 # If HOME_MODE is not set, the value of UMASK is used to create the mode.
-#HOME_MODE	0700
+# sshd `StrictModes yes` default does not allow group writable $HOME, hence g=rx
+# samba usershare functionality requires o=x set
+HOME_MODE	0751
 
 #
 # Password aging controls:
@@ -216,7 +218,7 @@ UMASK		022
 #
 PASS_MAX_DAYS	99999
 PASS_MIN_DAYS	0
-PASS_MIN_LEN	5
+PASS_MIN_LEN	8
 PASS_WARN_AGE	7
 
 #
@@ -233,7 +235,7 @@ SU_WHEEL_ONLY	no
 UID_MIN			 1000
 UID_MAX			60000
 # System accounts
-SYS_UID_MIN		  101
+SYS_UID_MIN		  201
 SYS_UID_MAX		  999
 # Extra per user uids
 SUB_UID_MIN		   100000
@@ -246,7 +248,7 @@ SUB_UID_COUNT		    65536
 GID_MIN			 1000
 GID_MAX			60000
 # System accounts
-SYS_GID_MIN		  101
+SYS_GID_MIN		  201
 SYS_GID_MAX		  999
 # Extra per user group ids
 SUB_GID_MIN		   100000
@@ -329,7 +331,7 @@ CHFN_RESTRICT		rwh
 # Note: If you use PAM, it is recommended to use a value consistent with
 # the PAM modules configuration.
 #
-#ENCRYPT_METHOD DES
+ENCRYPT_METHOD YESCRYPT
 
 #
 # Only works if ENCRYPT_METHOD is set to SHA256 or SHA512.
